pkg/runtime/program.go

```diff
@@ -92,3 +111,29 @@ func (p *Program) MustRun(ctx context.Context, setters ...Option) []byte {
 
 	return out
 }
+
+func (p *Program) validateParams(opts *Options) error {
+	if len(p.params) == 0 {
+		return nil
+	}
+
+	var missedParams []string
+
+	for n := range p.params {
+		_, exists := opts.params[n]
+
+		if !exists {
+			if missedParams == nil {
```

Comment:
There are no reasons to check `missedParams == nil` because `append(nil-slice, elem)` returns `slice{elem}` and doesn't panic

Reply:
Good to know, thanks!

Reply:
But, won't it re-allocate a new slice on every append?

Reply:
It will but not on every append. Finally, there is nothing prevents you from doing 
```go
missedParams :=  make([]string, 0, len(p.params))

for n := range p.params { ...
```

Reply:
That's what I'm doing, but lazily. Why do I need to allocate it if there are might be no errors?
There is a chance that escape analysis will optimize the code though...

Reply:
If you doing this because there are might be no errors, it's OK but need a comment. Otherwise it looks like an unnecessary check.

Reply:
Added a comment